src/components/ErrorBox.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import Radium from 'radium';
+import {connect} from 'react-redux';
+import {styles} from './styles';
+
+export class ErrorBox extends React.Component {
+  render() {
+    switch(false) {
+      case typeof this.props.loadApiError === "undefined":
+        return (
+          <span style={styles.errorBox}>
+              {this.props.loadApiError.message}
+          </span>
+        );
+      case typeof this.props.runCodeError === "undefined":
+        return (
+          <span style={styles.errorBox}>
+              {this.props.runCodeError.message}
+          </span>
+        );
+      case typeof this.props.googleDriveError === "undefined":
+        return (
+          <span style={styles.errorBox}>
+              {this.props.googleDriveError.message}
+          </span>
+        );
+      default:
+        return null; //if no error recorded in state do nothing
+    }
+  }
```

Comment:
I ended up implementing it like this instead.

```export class ErrorBox extends React.Component {
  render() {
    if (!(this.props.error === null || this.props.error === undefined)) {
      return <span style={styles.errorBox}>{this.props.error.message}</span>;
    } else {
      return null;
    }
  }
}```

Without first weeding out the nulls and undefined that can be returned by the selector you get errors about trying to read .message of null or undefined. This could also be implemented in the selector, but I just did it this way for now. What do you think